otter/log/spec.py

```diff
@@ -0,0 +1,118 @@
+"""
+Format logs with specification
+"""
+
+from copy import deepcopy
+
+from otter.constants import ServiceType
+from otter.log.formatters import IGNORE_FIELDS
+
+
+class UUID(basestring):
+    """
+    UUID
+    """
+
+
+class IPAddress(basestring):
+    """
+    IP Address
+    """
+
+
+# Allowed fields with their types
+fields = {
+    "num_servers": int,
+    "server_id": UUID,
+    "clb_id": int,
+    "ip_address": IPAddress,
+}
+
+
+# Fields that will occur in error
+error_fields = {
```

Comment:
I was also wondering if it's useful to tag things as API errors vs non-api errors?  We may be able to find bugs more easily in our code by filtering away the API errors?  (e.g. having an exception type or error type name be a field?)